discojs/discojs-core/src/models/gpt/index.ts

```diff
@@ -73,6 +38,7 @@ export class GPT extends Model {
     epochs = 1,
     tracker = new Sink()
   ): AsyncGenerator<EpochLogs, void> {
+    this.model.compile() // init the optimizer
```

Comment:
move it to the constructor, otherwise, it will re-init the optimizer every time someone wants to train (seems weird to me).